### DISCUSSION TITLE
is there a method to convert a parsed text back to text?

### DISCUSSION THREAD

**Discussion by ophiry** [explosion/spaCy#7334] [Help: Other Questions] [answered]
my intention is to parse a sentence, modify the parse tree, and regenerate the text from the modified tree.
(if there are multiple representations, getting all the options is preferred)

**Comment by adrianeboyd** [comment, marked as answer]
No, spacy probably isn't the right library for this. You can modify the parse tree keeping all tokens in place, but the underlying text is fixed for a particular `Doc`, so you can't reorder tokens by modifying the parse tree.

**Reply by ophiry** [reply]
Thanks,
are you aware of alternative tools, or papers \ research on this subject?
